hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryWorkflowImpl.java

```diff
@@ -206,9 +212,10 @@ public void handleQuery(
             TransactionBody txBody = null;
             if (paymentRequired) {
                 // 3.i Validate CryptoTransfer
-                allegedPayment = queryHeader.payment();
+                allegedPayment = queryHeader.paymentOrElse(Transaction.DEFAULT);
```

Comment:
Is using the default values really the right behavior here, if these fields were not set?

Reply:
I don't know if it's the preferred behavior, but doesn't using a default value here preserve what the previous code was already doing?

Reply:
Modified to throw instead of assigning the default